SAMtemplates/lambda_resources.yaml

```diff
@@ -328,11 +374,80 @@ Resources:
               - lambda:InvokeFunction
             Resource:
               - !GetAtt SplunkDeliveryStreamProcessor.Arn
+  ##################################################
+  # Certificate Checker
+  ##################################################
+  LambdaManagedPolicy:
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      Roles:
+        - !GetAtt ExpiryFunctionResources.Outputs.LambdaRoleName
+      PolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Action:
+              - secretsmanager:GetSecretValue
+            Resource:
+              - !ImportValue account-resources:ClinicalTrackerCACertSecret
+              - !ImportValue account-resources:ClinicalTrackerClientCertSecret
+              - !ImportValue account-resources:ClinicalTrackerClientSandboxCertSecret
+              - !ImportValue account-resources:PfpCACertSecret
+              - !ImportValue account-resources:PfpClientCertSecret
+              - !ImportValue account-resources:PfpClientSandboxCertSecret
+              - !ImportValue account-resources:SpinePublicCertificate
+          - Effect: Allow 
+            Action:
+             - secretsmanager:BatchGetSecretValue    
+            Resource: "*"
+  ExpiryFunctionResources:
+    Type: AWS::Serverless::Application
+    Properties:
+      Location: common_lambda_resources.yml
+      Parameters:
+        CloudWatchKMSKey: !GetAtt CloudWatchKMSKey.Arn
+        SplunkSubscriptionFilterRole: !GetAtt SplunkSubscriptionFilterRole.Arn
+        SplunkDeliveryStream: !GetAtt SplunkDeliveryStream.Arn
+        LambdaName: !Sub "${AWS::StackName}-CertificateChecker"
+        LogRetentionDays: !Ref LogRetentionDays
+        EnableSplunk: true
+
+  CertExpiryCheckFunction:
+    Type: AWS::Serverless::Function
+    Properties:
+      FunctionName: !Sub "${AWS::StackName}-CertificateChecker"
+      CodeUri: ../packages/
+      Handler: certificateChecker.handler
+      Runtime: nodejs18.x 
+      Role: !GetAtt ExpiryFunctionResources.Outputs.LambdaRoleArn
```

Comment:
can you add the following - it makes it easier to find the line that caused unexpected errors
```
    Environment:
      Variables:
        NODE_OPTIONS: --enable-source-maps
```